### ISSUE TITLE
[istio-rbac-breaks-non-http-listeners] Istio RBAC ServiceRoleBinding breaks non-http listeners

### ISSUE THREAD

**Issue by yanniszark** [istio/istio#24144]
**Bug description**

In Kubeflow, we have the following ServiceRoleBinding to allow only requests with the correct headers to access services in a namespace:

```yaml
apiVersion: rbac.istio.io/v1alpha1
kind: ServiceRoleBinding
metadata:
  annotations:
    role: admin
    user: user
  creationTimestamp: "2020-03-16T08:07:38Z"
  generation: 1
  name: owner-binding-istio
  namespace: kubeflow-user
  ownerReferences:
  - apiVersion: kubeflow.org/v1
    blockOwnerDeletion: true
    controller: true
    kind: Profile
    name: kubeflow-user
    uid: 33c2a0ef-675d-11ea-90b3-42010a84002e
  resourceVersion: "7939"
  selfLink: /apis/rbac.istio.io/v1alpha1/namespaces/kubeflow-user/servicerolebindings/owner-binding-istio
  uid: 33c8554c-675d-11ea-90b3-42010a84002e
spec:
  roleRef:
    kind: ServiceRole
    name: ns-access-istio
  subjects:
  - properties:
      request.headers[kubeflow-userid]: user
```

Istio assumes a port's protocol based on its name prefix (e.g., http, https, http2, grpc, etc.).
When a non-HTTP port is present at a Pod, the sidecar's listener will fail to start with the following error:
```
[2020-05-27 12:18:58.828][17][warning][config] [bazel-out/k8-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:87] gRPC config stream closed: 13,
[2020-05-27 12:18:59.229][17][warning][config] [external/envoy/source/common/config/grpc_mux_subscription_impl.cc:81] gRPC config for type.googleapis.com/envoy.api.v2.Listener rejected: Error adding/updating listener(s) 172.17.0.73_8443: Found header(name: "kubeflow-userid"
exact_match: "user"
) rule,not supported by RBAC network filter, virtualInbound: Found header(name: "kubeflow-userid"
exact_match: "user"
) rule,not supported by RBAC network filter
```

**What happened:**

Istio is trying to apply HTTP-filter specific configuration to an RBAC Network Filter (network filters act on TCP packets).

[X] Configuration Infrastructure
[X] Policies and Telemetry
[X] Security

**Expected behavior**

I would expect Istio to strip the HTTP-specific options when it is generating a network filter. This 
 bug means that a user has to only use http ports if they want to use HTTP-specific RBAC properties.

**Steps to reproduce the bug**

1. Enable Istio RBAC
2. Create a ServiceRoleBinding referencing headers.
3. Start a workload with an Istio Sidecar and a non-http port.
4. See logs of Istio sidecar.

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
```
client version: 1.3.1
cluster-local-gateway version: 
citadel version: release-1.3-20200214-10-15
galley version: release-1.3-20200214-10-15
ingressgateway version: release-1.3-20200214-10-15
nodeagent version: 
pilot version: release-1.3-20200214-10-15
policy version: release-1.3-20200214-10-15
sidecar-injector version: release-1.3-20200214-10-15
telemetry version: release-1.3-20200214-10-15
```
```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"clean", BuildDate:"2019-06-06T01:44:30Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"clean", BuildDate:"2019-06-06T01:36:19Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```

**How was Istio installed?**
These manifests: 

**Environment where bug was observed (cloud vendor, OS, etc)**
[MiniKF](https://www.kubeflow.org/docs/started/workstation/getting-started-minikf/)
OS: 
```
NAME="Ubuntu"
VERSION="16.04.6 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.6 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```

**Comment by incfly** [comment]
Old authz policy is deprecated. Please use the new AuthorizationPolicy. And reopen new bug if you still see issue or can not be even configured.

**Comment by yangminzhu** [comment]
note this bug should be addressed in 1.5 which is the last version that supports the ServiceRole and ServiceRoleBinding.

**Comment by andre-lx** [comment]
I'm getting exactly the same issue, with kubeflow, and this exact versions, using dex. Did you came with some solutions @yanniszark ?? Thanks
